run.ts

```diff
@@ -63,37 +63,34 @@ const launchDocker: SearchEngineLauncherFunction = async ({
   port,
   options,
 }) => {
-  const Image =
-    engine === 'elasticsearch'
-      ? 'elastic/elasticsearch:8.6.2'
-      : 'opensearchproject/opensearch:2.11.0'
-  console.log('Launching Docker container', Image)
-  const docker = new Dockerode()
-  const container = await docker.createContainer({
-    Env: [...options, 'path.data=/var/lib/search', 'path.logs=/var/log/search'],
-    HostConfig: {
-      AutoRemove: true,
-      Mounts: [
-        { Source: dataDir, Target: '/var/lib/search', Type: 'bind' },
-        { Source: logsDir, Target: '/var/log/search', Type: 'bind' },
-      ],
-      PortBindings: {
-        [`${port}/tcp`]: [{ HostIP: '127.0.0.1', HostPort: `${port}` }],
-      },
-    },
-    Image,
+  const subprocess = fork(
+    './node_modules/@nasa-gcn/architect-plugin-search/launchSearch.js',
+    [dataDir, logsDir, engine, port, options] as ForkOptions
```

Comment:
Why is this cast necessary?

Reply:
typescript
<img width="743" alt="Screenshot 2024-06-06 at 11 36 43 AM" src="https://github.com/nasa-gcn/architect-plugin-search/assets/6545874/71d65d32-a92b-4f64-83fb-d8c7d35681e6">

Reply:
TypeScript is pointing out a genuine error here. There are actually two reasons why these types are not compatible:

1. The command-line arguments to the child process should be an array of strings. `dataDir`, `logsDir`, and `engine` are fine, but `port` is a `number` and `options` is a `string[]`. Neither of the latter two types are permissible here.
2. `ForkOptions` is the type for the optional `option` object argument for `fork`.